### PR TITLE
added lesson 11: argparse

### PR DESCRIPTION
**Pull request by 3tom**
Dear @tomekDiakowTieto ,

Please take a look.

Thanks,
Tomasz

### CODE REVIEW

File: students/medrek_tomasz/lesson_11_argparse/regex_search.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    for file in os.listdir(dir_path):
+        if file.endswith(".txt"):
+            count_files += 1
+
+            try:
+                fh = open(os.path.join(dir_path, file), 'r')
+                string_from_file = fh.readlines()
+                fh.close()
+            except (UnicodeDecodeError, PermissionError) as e:
```
[the remark this answers is not in the document]
Comment (3tom) [reply students/medrek_tomasz/lesson_11_argparse/regex_search.py]: https://github.com/jedzej/tietopythontraining-basic/issues/606

File: students/medrek_tomasz/lesson_11_argparse/selective_copy.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    return(results.source, results.destination, results.extension)
+
+
+def main():
+
+    source, destination, file_extension = check_arg(sys.argv[1:])
+
+    try:
+        os.makedirs(destination)
+    except PermissionError as e:
```
[the remark this answers is not in the document]
Comment (3tom) [reply students/medrek_tomasz/lesson_11_argparse/selective_copy.py]: https://github.com/jedzej/tietopythontraining-basic/issues/606

File: students/medrek_tomasz/lesson_11_argparse/selective_copy.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            raise
+
+    copied_files = 0
+    for directory, _, files in os.walk(source):
+        for file in files:
+            if file.endswith(file_extension):
+                try:
+                    print(shutil.copy(os.path.join(directory, file),
+                                      destination))
+                except PermissionError as perm_err:
```
[the remark this answers is not in the document]
Comment (3tom) [reply students/medrek_tomasz/lesson_11_argparse/selective_copy.py]: https://github.com/jedzej/tietopythontraining-basic/issues/606

File: students/medrek_tomasz/lesson_11_argparse/selective_copy.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    source, destination, file_extension = check_arg(sys.argv[1:])
+
+    try:
+        os.makedirs(destination)
+    except PermissionError as e:
+        print(e)
+        exit()
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise
```
Comment (tomekDiakowTieto) [inline students/medrek_tomasz/lesson_11_argparse/selective_copy.py]: raise what :) ?

**Pull request merged**